purchases/src/main/kotlin/com/revenuecat/purchases/UpgradeInfo.kt

```diff
@@ -5,13 +5,14 @@ import com.android.billingclient.api.BillingFlowParams
 /**
  * This object holds the information used when upgrading from another sku.
  * @property oldSku The oldSku to upgrade from.
- * @property prorationMode The [BillingFlowParams.ProrationMode] to use when upgrading the given oldSku.
+ * @property replacementMode The [BillingFlowParams.SubscriptionUpdateParams.ReplacementMode] to use when upgrading the
+ * given oldSku.
  */
 @Deprecated(
```

Comment:
IMO yes, I think we already have versions of these methods using replacement mode, so we shouldn't have both I think. So we should remove them. Happy to chat more about it when I'm back!

Reply:
To clarify, it's about removing methods like this one, taking an `UpgradeInfo`: 
```kotlin
fun purchasePackage(
    activity: Activity,
    packageToPurchase: Package,
    upgradeInfo: UpgradeInfo,
    callback: ProductChangeCallback,
)
``` 
In this PR, `UpgradeInfo` now contains a `GoogleReplacementMode` instead of a `GoogleProrationMode`. As far as I'm aware, there are no methods taking a `GoogleReplacementMode`/`GoogleProrationMode` directly. 
  
(I'm not arguing for or against removing them btw, just clarifying 🙂)

Reply:
Yes, I agree we should take the opportunity to clean all those up and only leave the `purchase` function that takes `PurchaseParams.Builder`

Happy to help out, since we have more context, and there are a loot of them

Reply:
Makes sense! I'll take a stab at it. Nice codebase-exploration opportunity haha

Reply:
Closing the loop on this: https://github.com/RevenueCat/purchases-android/pull/1711